test/common/trafficmanager/fakeprovider/endpoint.go

```diff
@@ -25,6 +25,12 @@ const (
 	Weight = int64(50)
 )
 
+var (
+	// returnEndpointForbiddenErr is to control whether to return forbidden error or not.
+	// Note: it's not thread safe.
+	returnEndpointForbiddenErr bool
```

Comment:
why not make it thread safe?

Reply:
all the tests run sequentially. will do in a separate PR.